app/controllers/articles_controller.rb

```diff
@@ -4,6 +4,10 @@ def index
   end
 
   def show
-    @article = Article.find(params[:id])
+    if params[:id].to_i > 0
```

Comment:
Hmmmm is this `to_i` necessary?, `find` will coerce the value using `to_i` as well. Is the intention here the same as calling `.is_a? Numeric`?.

It probably is necessary but I'm wondering when a non numeric ID would be passed in here :thinking: .

Reply:
There are 2 possibilities here: either the param is a number like `23`, or it's a slug like `hello_world`. I'm using `to_i` to distinguish between the two: either I look up that ID in the database, or I look for the article that has that slug in the database.